src/app/modules/data-set/annotation/annotation.component.ts

```diff
@@ -24,46 +25,41 @@ export class AnnotationComponent implements OnInit {
     Validators.min(0),
     Validators.max(3),
   ]);
-  public heuristics = new FormControl();
 
   private availableHeuristics: Map<string, string[]> = new Map();
-
-  public applicableHeuristics: Map<string, string> = new Map();
+  public heuristicsAndReasons: Map<string, string> = new Map();
   public annotatorId: string = '';
   private isHeuristicReasonChanged: boolean = false;
+  private warningSnackbarOptions: any = {horizontalPosition: 'center', verticalPosition: 'bottom', duration: 3000, panelClass: ['warningSnackbar']};
+  private successSnackBarOptions: any = {horizontalPosition: 'center', verticalPosition: 'bottom', duration: 3000, panelClass: ['successSnackbar']};
+  private errorSnackBarOptions: any = {horizontalPosition: 'center', verticalPosition: 'bottom', duration: 3000, panelClass: ['errorSnackbar']};
 
   @Output() newAnnotation: EventEmitter<Annotation> = new EventEmitter<Annotation>();
   @Output() changedAnnotation: EventEmitter<Annotation> = new EventEmitter<Annotation>();
 
-  constructor(private annotationService: AnnotationService, private changeDetector: ChangeDetectorRef) { }
+  constructor(private annotationService: AnnotationService, private changeDetector: ChangeDetectorRef, private _snackBar: MatSnackBar) { }
 
   public ngOnInit(): void {
     this.annotatorId = this.previousAnnotation?.annotator.id + '';
-    if (!this.disableEdit) {
-      this.annotationService.getAvailableHeuristics().subscribe(res => this.initHeuristics(res, this.availableHeuristics));
-    } else if (this.previousAnnotation) {
-      let previousHeristics = this.previousAnnotation.applicableHeuristics.map(h => h.description);
-      this.availableHeuristics.set(this.codeSmell, previousHeristics);
-    }
+    this.annotationService.getAvailableHeuristics().subscribe(res => this.initHeuristics(res, this.availableHeuristics));
+    if (this.previousAnnotation) this.setupInputFromPreviousAnnotation();
   }
 
   public ngOnChanges(): void {
     this.severityFormControl.setValue(0);
     if (this.disableEdit) {
       this.severityFormControl.disable();
-      this.heuristics.disable();
     }
-    this.heuristics.setValue([]);
-    this.applicableHeuristics = new Map();
+    this.heuristicsAndReasons = new Map();
     this.setupInputFromPreviousAnnotation();
   }
 
   public ngAfterViewChecked(): void {
     if (this.isHeuristicReasonChanged) {
-      for (let description of this.applicableHeuristics.keys()) {
-        let reasonInput = document.getElementById('reason-' + description + '-' + this.annotatorId) as HTMLInputElement;
-        if (reasonInput) {
-          reasonInput.value = this.applicableHeuristics.get(description)!;
+      for (let heuristic of this.heuristicsAndReasons.keys()) {
+        let reason = document.getElementById('reason-' + heuristic + '-' + this.annotatorId) as HTMLInputElement;
```

Comment:
This looks weird. Using the "document" and manipulating the DOM directly in components is an angular application code smell. What are you trying to accomplish here and is there a more angular-friendly way to do this (e.g., through ngModel)?